### PR TITLE
Env

### PR DESCRIPTION
**Pull request by DIWAKARKASHYAP**
## Problem
Resolves  https://github.com/langfuse/langfuse/issues/232

## Changes

1- Add conditional statements in the constructors of both the langfuse and langfuseWeb classes located in the langfuse-js/langfuse/src/langfuse.ts file
2- Rename the previous environment variable names to the new ones: LANGFUSE_PUBLIC_KEY, LANGFUSE_SECRET_KEY, and LANGFUSE_HOST

### Bump level

- [ ] Major
- [x] Minor
- [ ] Patch

### Libraries affected

- [ ] All of them
- [x] langfuse
- [ ] langfuse-node

### Changelog notes

- Added support for X

### CODE REVIEW

File: langfuse/src/langfuse.ts [hunk 1]
```diff
@@ -22,9 +25,33 @@ export class Langfuse extends LangfuseCore {
...
+    }
+
+    if (!params.secretKey && process.env.LANGFUSE_SECRET_KEY) {
+      params.secretKey = process.env.LANGFUSE_SECRET_KEY
+    }
+
+    if (!params.baseUrl && process.env.LANGFUSE_HOST) {
+      params.baseUrl = process.env.LANGFUSE_HOST
+    }
+
```
Comment (DIWAKARKASHYAP) [inline langfuse/src/langfuse.ts:50]: i am confuse a bit , from
 ```ts
 if (!params.publicKey && process.env.LANGFUSE_PUBLIC_KEY) {
      params.publicKey = process.env.LANGFUSE_PUBLIC_KEY
    }

  if (!params.secretKey && process.env.LANGFUSE_SECRET_KEY) {
      params.secretKey = process.env.LANGFUSE_SECRET_KEY
    }

  if (!params.baseUrl && process.env.LANGFUSE_HOST) {
      params.baseUrl = process.env.LANGFUSE_HOST
    }
//☝️  if the user provides credentials in both the environment variables and the constructor, the code gives priority to the constructor values, 
```
to 

```ts

    params.publicKey =  process.env.LANGFUSE_PUBLIC_KEY || params.publicKey
    params.secretKey =  process.env.LANGFUSE_SECRET_KEY || params.secretKey
    params.baseUrl =  process.env.LANGFUSE_HOST || params.baseUrl;

//☝️ This code is now more readable. The key difference is that if the user provides credentials in both the environment variables and the constructor, the code gives priority to the environment variables, 
```

please read the comment 

 i can do the same in langfuseWeb

File: langfuse/src/langfuse.ts [hunk 2]
```diff
@@ -8,6 +8,9 @@ import {
 import { type LangfuseStorage, getStorage } from "./storage";
 import { version } from "../package.json";
 import { type LangfuseOptions } from "./types";
+import dotenv from 'dotenv';
+
+dotenv.config();
```
Comment (maxdeichmann) [inline langfuse/src/langfuse.ts:13]: Will this here work in all the environments like Browsers or different Node versions?

File: langfuse/src/langfuse.ts (hunk 1 above)
Comment (maxdeichmann) [reply langfuse/src/langfuse.ts:50]: Hi, thanks for the comment. We should give priority to the constructor variables.

Comment (maxdeichmann) [reply langfuse/src/langfuse.ts:50]: In this case, do we even need L39-L49?

**Comment by marcklingen** [comment]
@DIWAKARKASHYAP 
Thank you for your contribution. Great to include this in langfuse and not langfuse-core to not affect langfuse-web implementation where envs are not available.

Some folks use the langfuse package in Cloudflare workers or Deno. In both, process.env is not available. You used the `dotenv` package, but did not install it. Is there a way to use environment variables only in node and for all other environments to not have this crash?

File: langfuse/src/langfuse.ts (hunk 2 above)
Comment (DIWAKARKASHYAP) [reply langfuse/src/langfuse.ts:13]: i test this on node version
    18.18.0
    18.13.0
    16.14.2
    16.0.0 
    14.17.0
    12.11.1
it works perfectly

File: langfuse/src/langfuse.ts (hunk 1 above)
Comment (DIWAKARKASHYAP) [reply langfuse/src/langfuse.ts:50]: I think yes. Beacuse If we do not check the PK and SK in the environment variable, how can we use them? What you think ?

**Comment by ksjogo** [comment]
There actually is a getEnv utility function in the core package.
See #50 
One could probably just use that instead.

**Comment by marcklingen** [comment]
I close this as this was implemented in #90

**Pull request closed without merging**